tiledb/common/thread_pool/thread_pool.h

```diff
@@ -136,7 +370,7 @@ class ThreadPool {
    * @return Status::Ok if all tasks returned Status::Ok, otherwise the first
    * error status is returned
    */
-  Status wait_all(std::vector<Task>& tasks);
+  Status wait_all(std::vector<ThreadPoolTask*>& tasks);
```

Comment:
Methods that take a `std::vector<ThreadPoolTask*>` should not be public.

Reply:
why?

Reply:
The lifetime of the `ThreadPoolTask*` pointers is not clear from the signature. If we want in the future to wait for multiple types of tasks at the same time we can bring it back.

Reply:
Fair enough.